charts/cloudzero-agent/templates/deploy.yaml

```diff
@@ -30,6 +30,36 @@ spec:
       priorityClassName: "{{ .Values.server.priorityClassName }}"
 {{- end }}
       serviceAccountName: {{ template "cloudzero-agent.serviceAccountName.server" . }}
+      initContainers:
```

Comment:
I think one potential downside of running this as an initcontainer and not just another container is that the prometheus agent container won't start until this validation is done, right? which I think is totally fine, but I wonder what the difference there is in startup time (if any)

Reply:
Yes that is correct. Last night during some of the work I came to the following realization:
1. If we set (restartPolicy: Never) which makes it available as true sidecar, we need to make sure the container continues to run (unless I misunderstand what I saw in my experimenting). This didn't seem great to me.
2. Another approach would be to be an init container, and create a script and pyenv which we then use as a pod `pre/post` **lifecycle** hook. I think this is a potential good way to go as we don't have a container running all the time (also not sure how much of an issue that is either). 

WRT startup, I saw only a smallish start difference basically while awaiting the start of the KMS. So it was almost negligible impact on the start time.